addons/website_quote/models/sale_order.py

```diff
@@ -77,6 +77,11 @@ def _compute_amount_undiscounted(self):
             total += line.price_subtotal + line.price_unit * ((line.discount or 0.0) / 100.0) * line.product_uom_qty  # why is there a discount in a field named amount_undiscounted ??
         self.amount_undiscounted = total
 
+    @api.onchange('partner_id')
+    def onchange_partner_id(self):
+        super(SaleOrder, self).onchange_partner_id()
+        self.note = self.template_id.note or self.note
```

Comment:
It will not works if the quotation template is selected after the customer... doesn't work with default template or whenever you change the customer.
In specs this point added by fgi on review.